public/app/features/templating/templateSrv.js

```diff
@@ -154,6 +154,13 @@ function (angular, _, kbn) {
           }
         }
 
+        var adhocVar = _.find(self._adhocVariables, function(v) {
```

Comment:
this should not be here, the filter should be added to the query automatically in the datasource query code (see elasticsearch or InfluxDB for example)